dvc/updater.py

```diff
@@ -48,7 +48,35 @@ def _with_lock(self, func, action):
             logger.debug(msg.format(self.lock.lockfile, action))
 
     def check(self):
-        if os.getenv("CI") or env2bool("DVC_TEST") or PKG == "snap":
+        if os.getenv("CI") or env2bool("DVC_TEST"):
+            return
+
+        if PKG == "snap":
+            # Ideally need to check if SNAP_CHANNEL.startswith(self.current),
+            # where SNAP_CHANNEL is:
+            #   snap info dvc | grep -E '^tracking:' | awk '{print $2}'
+            # However a quicker way is to skip if a user-configurable variable
+            # `skip_update_check` is set
+            import subprocess
+
+            kwargs = {"close_fds": True}
+            if os.name == "nt":
+                kwargs["shell"] = True
+                cmd = []
+            else:
+                kwargs["shell"] = False
+                cmd = ["/bin/sh", "-c"]
+            p = subprocess.Popen(
+                cmd + ["snapctl", "get", "skip_update_check"], **kwargs
```

Comment:
Not a fan of creating processes on `updater`.

Reply:
Also, this runs on every dvc command in the same process, so this will turn ugly slow.

Reply:
yes I don't like it either but I don't see any other way apart from having two different 0.x releases: one which pushes to `v0/stable` without any warning logic, and one which for the next 30 days pushes to `latest/stable` with the warning but none of these `subprocess` checks.

Reply:
Maybe it is not worth having a special notification. YOu've created separate channels that will make sure everything is setup properly in the future, so we could afford some people automatically upgrading to 1.0 here.

Reply:
I don't know - based on discord it looked like at least one person had errors from an automatic upgrade before we rolled back.

Reply:
removed